src/SearchIndex/util.ts

```diff
@@ -167,7 +169,7 @@ export function convertTaxonomyToTrie(taxonomy: Taxonomy): TrieFacet {
     name: '',
   };
 
-  function addToRes(entityList: TaxonomyEntity[], ref: { [key: string]: any }, property: string) {
+  function addToRes(entityList: TaxonomyEntity[], ref: RefObj, property: string) {
```

Comment:
think we can replace RefObj with `TrieFacet` and refer to re-assigned `ref` `as Triefacet` and have the entity declared as one as well `entity: Triefacet = {}`